### DISCUSSION TITLE
Support for VisionOS

### DISCUSSION THREAD

**Discussion by phiren** [airbnb/lottie-ios#2096] [General]
Project not building even for Phone target due to UIScreen & Haptic Feedback API.
Small fixes will make whole library usable for VisionOS.

**Comment by phiren** [comment]
It taken just 10 mins to fix and run code. 
It will be really good if support provided in master repository.

**Reply by quanticapps** [reply]
Hi, can you please tell us how you fix it as we're trying to run our project on Vision using Lottie? Thanks very much

**Comment by calda** [comment]
PRs welcome!

**Reply by phiren** [reply]
Ok, created PR.

**Reply by quanticapps** [reply]
Hi @phiren since the Xcode 15 beta 6 when building visionOS apps, #if os(iOS) is no longer considered part of the project. This behavior is different from beta 1 to beta 5. Can you please add the os(xrOS) such as in Data+Compression.swift line 116 etc ? Thank you

**Comment by calda** [comment]
I'm taking a look at this and will hopefully include visionOS support in the next release.

**Reply by calda** [reply]
There are some complications that will need to be figured out around continuing to support old versions of Xcode that don't include the visionOS SDK yet, especially wrt the precompiled XCFramework used in lottie-spm.

**Reply by calda** [reply]
here's a PR: https://github.com/airbnb/lottie-ios/pull/2152

**Comment by calda** [comment]
Support for visionOS has been merged to master and will be included in the next release.

**Reply by calda** [reply]
The Xcode 15 release candidate doesn't include the visionOS SDK. Since the XCFramework used by the lottie-spm package will need to be built by a non-beta build, lottie-spm won't be able to support visionOS for now. If you need visionOS support, use the lottie-ios repo directly for now.

**Reply by calda** [reply]
Lottie 4.3.0 has been released and includes support for visionOS: https://github.com/airbnb/lottie-ios/discussions/2189

**Reply by tracyhenry** [reply]
When I install using the swift package manager with "https://github.com/airbnb/lottie-spm.git", it still doesn't work. Installing with "lottie-ios.git" worked.

**Reply by calda** [reply]
Like [mentioned above](https://github.com/airbnb/lottie-ios/discussions/2096#discussioncomment-6983992), we won't be able to support visionOS in the lottie-spm repo until the visionOS SDK is officially released in a production version of Xcode. Until then if you need support for visionOS please use the lottie-ios repo directly.
